### ISSUE TITLE
creating subnetwork from intact network missing nodes. 

### ISSUE THREAD

**Issue by risserlin** [EBI-IntAct/IntActApp#6]
I run the following commands in R (I have done the same actions from inside cytoscape and get the same result):

intact_command <- "intact query seedTerms='P49418 Q8N729 Q5VZR2 Q8WYP3 Q3SY00 Q9NPI5 Q7RTP6 A2A288 A1X283 P52849 Q717R9 Q96MF6 Q8N158 Q08345 Q5JSZ5 Q6ZSJ9 A8MTA8 Q86VR8 Q9BWX5 P10826 Q8IY33 Q9UQ88 O14545 Q9Y3S1 O43295 Q6NSI4 Q6J9G0 Q8TB24 Q96J02 Q9H1Q7 Q99973 P08151 O60890 Q8NBV8 Q8TD84 Q6GQQ9 O15240 Q9HBG7 Q8WZ71 Q2KJY2 Q14160 Q9UHG0 H7BZ55 Q02246 Q6ZRS2 Q9Y4E6 O95180 Q13177 Q6ZNL6 Q9UMN6 Q9UK61 Q14118 P50570 Q9Y6N7 O60486 P98160 Q96RT6 Q14183 Q8IXW0 Q93074 Q8N4Y2 Q5U5Q3 Q9HCK4 Q8WWW0 Q9HCK1 Q5T9C2 P46013 Q6ZU52 Q9H427 O43426 O43184 P57058 P54259 Q8WWN8 Q9HCI5 Q08493 Q96T23 Q14624 Q99743 Q96AY4 Q96F05 O15211 Q6UXB0 Q7Z6E9 Q9UJY5 Q8NDB2 O94875 Q8WXE0 Q6F5E8 Q8WUF5 Q8NI99 Q6ZS10 Q12756 Q5SYB0 Q9UPS6 Q96DX4 Q7Z7G1'"

commandsRun(intact_command)

intact_genes <- c("P49418", "Q8N729", "Q5VZR2", "Q8WYP3", "Q3SY00", "Q9NPI5", "Q7RTP6", "A2A288", "A1X283", "P52849", "Q717R9", "Q96MF6", "Q8N158", "Q08345","Q5JSZ5", "Q6ZSJ9", "A8MTA8", "Q86VR8", "Q9BWX5", "P10826", "Q8IY33", "Q9UQ88", "O14545", "Q9Y3S1", "O43295", "Q6NSI4", "Q6J9G0", "Q8TB24", "Q96J02", "Q9H1Q7", "Q99973", "P08151", "O60890", "Q8NBV8", "Q8TD84", "Q6GQQ9", "O15240", "Q9HBG7", "Q8WZ71", "Q2KJY2", "Q14160", "Q9UHG0", "H7BZ55", "Q02246", "Q6ZRS2", "Q9Y4E6", "O95180", "Q13177", "Q6ZNL6", "Q9UMN6", "Q9UK61", "Q14118", "P50570", "Q9Y6N7", "O60486", "P98160", "Q96RT6", "Q14183", "Q8IXW0", "Q93074", "Q8N4Y2", "Q5U5Q3", "Q9HCK4", "Q8WWW0", "Q9HCK1", "Q5T9C2", "P46013", "Q6ZU52", "Q9H427", "O43426", "O43184", "P57058", "P54259", "Q8WWN8", "Q9HCI5", "Q08493", "Q96T23", "Q14624", "Q99743", "Q96AY4", "Q96F05", "O15211", "Q6UXB0", "Q7Z6E9", "Q9UJY5", "Q8NDB2", "O94875", "Q8WXE0", "Q6F5E8", "Q8WUF5", "Q8NI99", "Q6ZS10", "Q12756", "Q5SYB0" ,"Q9UPS6", "Q96DX4", "Q7Z7G1")

intact_table_columns <- getTableColumns(table = "node")

nodes_of_interest_suid <- intact_table_columns$SUID[which(intact_table_columns$`IntAct::Preferred Id` %in% intact_genes)]

createSubnetwork(nodes_of_interest_suid)

The created subnetwork, although there are 86 nodes selected in the original intact network, only contains 28 nodes.  In the original network if you hide all unselected nodes you can see all the 86 selected nodes.  I am wondering what is special about the intact network that doesn't transfer over those nodes to the subnetwork.  When I run the same command with a psicquic interaction network it doesn't have this issue.

Thanks, 
Ruth

**Comment by EliotRagueneau** [comment]
Hi Ruth, 

There is in fact 86 nodes in the created subnetworks, but only 28 are visible, however, we chose to make subnetworks of IntAct networks proper IntAct networks themselves. 
It happens that in IntAct networks, we have some filters that are hiding all the nodes that are not connected to anything (no edges). We could eventually provide an option to disable this filter so that you can see those lonely nodes if you want.

By the way, you can observe the same result by telling that you don't want to include interacting partners, using the following query for instance:

```R
intact_command <- "intact query includeSeedPartners=false seedTerms='P49418 Q8N729 Q5VZR2 Q8WYP3 Q3SY00 Q9NPI5 Q7RTP6 A2A288 A1X283 P52849 Q717R9 Q96MF6 Q8N158 Q08345 Q5JSZ5 Q6ZSJ9 A8MTA8 Q86VR8 Q9BWX5 P10826 Q8IY33 Q9UQ88 O14545 Q9Y3S1 O43295 Q6NSI4 Q6J9G0 Q8TB24 Q96J02 Q9H1Q7 Q99973 P08151 O60890 Q8NBV8 Q8TD84 Q6GQQ9 O15240 Q9HBG7 Q8WZ71 Q2KJY2 Q14160 Q9UHG0 H7BZ55 Q02246 Q6ZRS2 Q9Y4E6 O95180 Q13177 Q6ZNL6 Q9UMN6 Q9UK61 Q14118 P50570 Q9Y6N7 O60486 P98160 Q96RT6 Q14183 Q8IXW0 Q93074 Q8N4Y2 Q5U5Q3 Q9HCK4 Q8WWW0 Q9HCK1 Q5T9C2 P46013 Q6ZU52 Q9H427 O43426 O43184 P57058 P54259 Q8WWN8 Q9HCI5 Q08493 Q96T23 Q14624 Q99743 Q96AY4 Q96F05 O15211 Q6UXB0 Q7Z6E9 Q9UJY5 Q8NDB2 O94875 Q8WXE0 Q6F5E8 Q8WUF5 Q8NI99 Q6ZS10 Q12756 Q5SYB0 Q9UPS6 Q96DX4 Q7Z7G1'"
```
You will see that the result includes only the 28 nodes you were obtaining in your method because when you directly query for nodes like this, you only receive the nodes that are in fact interacting with another protein in the set you provided.

To sum up,  all the 86 proteins you were querying for are inside IntAct and interacting with lots of the proteins, however, if you're looking at interactions that happen only between the 86 proteins you provided, only 28 of them will interact with another protein, and IntAct App only displays molecules that do interact with other molecules.

Please tell me which behaviour you would expect in those cases. Would you like an option to query and display all proteins in a set, no matter if they interact with something else or not?

**Comment by risserlin** [comment]
So truthfully I was only doing this because I couldn't get includeSeedPartners=false to work but now that that is working this is less of an issue.
But I am curious, how are you hiding those nodes?  I tried to use Select -> Unhide all hidden nodes and it didn't make those nodes appear.

**Comment by risserlin** [comment]
It would be helpful to have the ability to see all proteins even if they had no partners.  sometimes highlighting the genes that aren't represented can be useful.

**Comment by EliotRagueneau** [comment]
I don't know exactly why the unhide option doesn't work, but I believe it's because we're listening to all changes made on the network views, and we're refiltering afterwards. Or it might just be because I hid them programmatically and so we can not unhide them using menus, but I don't think it's that.

**Comment by EliotRagueneau** [comment]
> It would be helpful to have the ability to see all proteins even if they had no partners. sometimes highlighting the genes that aren't represented can be useful.

For the visualisation, it's quite easy to do as it is just a filter as the other ones, so I can include it's UI in the node menus to activate or deactivate it.

Regarding the direct queries, however, it implies a lot more changes in the back-end so I'm not sure we're gonna support that right away, but I will include that in our TO-DO list.

**Comment by dotasek** [comment]
HI Eliot,

Ruth suggested I chime in on the subject of hiding / unhiding objects via the API.

There are two ways of doing this that should allow users to later show the nodes: 

1. Setting the bypass NODE_VISIBLE to false via view.setLockedValue() (I use this in CyREST [here](https://github.com/cytoscape/cyREST/blob/a991b9a7085394b97bbe37fea92af7fce04e8a58/src/main/java/org/cytoscape/rest/internal/datamapper/VisualStyleMapper.java#L253))
2. Creating a passthrough mapping from a node data column (Used in CyREST [here](https://github.com/cytoscape/cyREST/blob/a991b9a7085394b97bbe37fea92af7fce04e8a58/src/main/java/org/cytoscape/rest/internal/datamapper/VisualStyleMapper.java#L138)). THIS WILL BE LESS INTUITIVE FOR THE USER. 'Show all...' will have no effect on this; to show the hidden objects, users, or the app need to remove the mapping. However, this also means that the data column will maintain which objects are to be visible, and the mapping can quickly be recreated.
